src/libraries/System.Net.Http.Json/tests/FunctionalTests/HttpClientJsonExtensionsTests.cs

```diff
@@ -47,45 +46,31 @@ public async Task TestGetFromJsonAsync()
                         per.Validate();
                     }
                 },
-                async server =>
-                {
-                    for (int i = 0; i < NumRequests; i++)
-                    {
-                        await server.HandleRequestAsync(content: json, headers: headers);
-                    }
```

Comment:
Why is that you only have to call `HandleRequestAsync` once now?

Reply:
@Jozkee In the previous implementation we actually had to create four responses from the socket-based loopback server, one for each request since the callback was only invoked once.

Now we invoke the callback for each request so this is no longer necessary.